lib/bolt/transport/local/shell.rb

```diff
@@ -67,8 +67,7 @@ def handle_sudo_errors(err, pid)
 
         def copy_file(source, dest)
           if source.is_a?(StringIO)
-            File.open("tempfile", "w") { |f| f.write(source.read) }
-            execute(['mv', 'tempfile', dest])
+            execute(['echo', source.read.to_s, '>', dest.to_s])
```

Comment:
Getting the escaping and interpolation just right here will require a bit more work than just a `to_s` method.

Reply:
Actually I'm thinking redirection just wont work due to https://tickets.puppetlabs.com/browse/BOLT-1056